target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8072-wax620.dts

```diff
@@ -0,0 +1,221 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+/dts-v1/;
+
+#include "ipq8074.dtsi"
+#include "ipq8074-hk-cpu.dtsi"
+#include "ipq8074-ess.dtsi"
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	model = "Netgear WAX620";
+	compatible = "netgear,wax620", "qcom,ipq8074";
+
+	aliases {
+		serial0 = &blsp1_uart5;
+		ethernet0 = &dp6;
+		led-boot = &led_system_blue;
+		led-failsafe = &led_system_red;
+		led-running = &led_system_green;
+		led-upgrade = &led_system_blue;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+		/*
+		 * Netgear's U-Boot adds "ubi.mtd=rootfs root=mtd:ubi_rootfs"
+		 * That fails to create a UBI block device, so add it here.
+		*/
+		bootargs-append = " ubi.block=0,rootfs root=/dev/ubiblock0_1";
```

Comment:
Is the partition called `rootfs`?

ubi.mtd is a valid option and should be preffered over setting the block manually.

Reply:
In that case, it is really weird that `ubi.mtd=rootfs` does not work.
Can you post the output of `ubinfo -a`?

Reply:
Here you go:
```
root@Netgear:~# ubinfo -a
UBI version:                    1
Count of UBI devices:           1
UBI control device major/minor: 10:126
Present UBI devices:            ubi0

ubi0
Volumes count:                           3
Logical eraseblock size:                 126976 bytes, 124.0 KiB
Total amount of logical eraseblocks:     480 (60948480 bytes, 58.1 MiB)
Amount of available logical eraseblocks: 0 (0 bytes)
Maximum count of volumes                 128
Count of bad physical eraseblocks:       0
Count of reserved physical eraseblocks:  40
Current maximum erase counter value:     74
Minimum input/output unit size:          2048 bytes
Character device major/minor:            246:0
Present volumes:                         0, 1, 2

Volume ID:   0 (on ubi0)
Type:        dynamic
Alignment:   1
Size:        39 LEBs (4952064 bytes, 4.7 MiB)
State:       OK
Name:        kernel
Character device major/minor: 246:1
-----------------------------------
Volume ID:   1 (on ubi0)
Type:        dynamic
Alignment:   1
Size:        72 LEBs (9142272 bytes, 8.7 MiB)
State:       OK
Name:        rootfs
Character device major/minor: 246:2
-----------------------------------
Volume ID:   2 (on ubi0)
Type:        dynamic
Alignment:   1
Size:        325 LEBs (41267200 bytes, 39.3 MiB)
State:       OK
Name:        rootfs_data
Character device major/minor: 246:3
```

Reply:
Ok, this is completely normal.
Can you post the full bootlog with `ubi.block` set and without?

Reply:
> Ok, this is completely normal. Can you post the full bootlog with `ubi.block` set and without?

Just change the dts and do a reboot? Or do I need to build a new image?
Change to this:
```
bootargs-append = " ubi.mtd=rootfs root=/dev/ubiblock0_1";
```

Reply:
I think I understand what you meant. I build a new image with the following:
```
chosen {
  stdout-path = "serial0:115200n8";
  /*
  * Netgear's U-Boot adds "ubi.mtd=rootfs root=mtd:ubi_rootfs"
  * That fails to create a UBI block device, so add it here.
  */
  bootargs-append = " root=/dev/ubiblock0_1";
};
```
```
root@Netgear:/# cat /sys/firmware/devicetree/base/chosen/bootargs-append
 root=/dev/ubiblock0_1root
```
[Boot log for only](https://gist.github.com/skramstad/79350cd8114701027f2c95a1f129d676) `root=/dev/ubiblock0_1root` set.

-------------------------
Below is the old log.

```
bootargs-append = " ubi.block=0,rootfs root=/dev/ubiblock0_1";
```
[This is the boot log with ubi set](https://gist.github.com/skramstad/df094be2993084a7400d97dc46912a3e).

Reply:
@robimarko did you want me to remove the whole line?
`bootargs-append = " root=/dev/ubiblock0_1";`

Reply:
Yes I think so too. Been running it since the posting. Thanks for helping out!